src/libraries/System.Runtime.InteropServices/src/System.Runtime.InteropServices.csproj

```diff
@@ -52,22 +53,22 @@
   </ItemGroup>
   <ItemGroup Condition="'$(TargetsUnix)' == 'true'">
     <Compile Include="System\Runtime\InteropServices\PosixSignalRegistration.Unix.cs" />
-    <Compile Include="$(CommonPath)Interop\Unix\System.Native\Interop.PosixSignal.cs"
-             Link="Common\Interop\Unix\Interop.PosixSignal.cs" />
-    <Compile Include="$(CommonPath)Interop\Unix\System.Native\Interop.InitializeTerminalAndSignalHandling.cs"
-             Link="Common\Interop\Unix\Interop.InitializeTerminalAndSignalHandling.cs" />
-    <Compile Include="$(CommonPath)Interop\Unix\Interop.Libraries.cs"
-             Link="Common\Interop\Unix\Interop.Libraries.cs" />
-    <Compile Include="$(CommonPath)Interop\Unix\Interop.Errors.cs"
-             Link="Common\CoreLib\Interop\Unix\Interop.Errors.cs" />
-    <Compile Include="$(CommonPath)Interop\Unix\Interop.IOErrors.cs"
-             Link="Common\Interop\Unix\Interop.IOErrors.cs" />
+    <Compile Include="$(CommonPath)Interop\Unix\System.Native\Interop.PosixSignal.cs" Link="Common\Interop\Unix\Interop.PosixSignal.cs" />
+    <Compile Include="$(CommonPath)Interop\Unix\System.Native\Interop.InitializeTerminalAndSignalHandling.cs" Link="Common\Interop\Unix\Interop.InitializeTerminalAndSignalHandling.cs" />
+    <Compile Include="$(CommonPath)Interop\Unix\Interop.Libraries.cs" Link="Common\Interop\Unix\Interop.Libraries.cs" />
+    <Compile Include="$(CommonPath)Interop\Unix\Interop.Errors.cs" Link="Common\CoreLib\Interop\Unix\Interop.Errors.cs" />
+    <Compile Include="$(CommonPath)Interop\Unix\Interop.IOErrors.cs" Link="Common\Interop\Unix\Interop.IOErrors.cs" />
   </ItemGroup>
   <ItemGroup Condition="'$(TargetsWindows)' == 'true'">
     <Compile Include="System\Runtime\InteropServices\PosixSignalRegistration.Windows.cs" />
+    <Compile Include="$(CommonPath)Interop\Windows\Kernel32\Interop.SetConsoleCtrlHandler.cs" Link="Common\Interop\Windows\Interop.SetConsoleCtrlHandler.cs" />
+    <Compile Include="$(CommonPath)System\IO\Win32Marshal.cs" Link="Common\System\IO\Win32Marshal.cs" />
+    <Compile Include="$(CommonPath)Interop\Windows\Interop.Libraries.cs" Link="Common\Interop\Windows\Interop.Libraries.cs" />
+    <Compile Include="$(CommonPath)Interop\Windows\Interop.Errors.cs" Link="Common\Interop\Windows\Interop.Errors.cs" />
+    <Compile Include="$(CommonPath)Interop\Windows\Kernel32\Interop.FormatMessage.cs" Link="Common\Interop\Windows\Interop.FormatMessage.cs" />
   </ItemGroup>
   <ItemGroup Condition="'$(TargetsBrowser)' == 'true'">
```

Comment:
This test project only has configurations for Windows, Unix, and Browser... I don't know why.
https://github.com/dotnet/runtime/pull/55333/files/27c743359ceb5be57a44708d4b8ac37502381f53#diff-f13d1871f8be09aa0e29f7ebf0e847e8dd2c0f7c62db5099542cefc945495d7eR5
@ViktorHofer?

Reply:
I guess that System.Runtime.InteropServices.csproj project did not need any special casing for Android and Apple devices OSes before the PosixSignal PR. If the PosixSignal APIs are meant to throw PNSE on these OSes, it does need the special casing now.  

I think it may be best to move the implementation of the signal APIs (or even the whole System.Runtime.InteropServices) into CoreLib. I can be separate PR.

Reply:
Ah, I see, I missed that Tom's PR updated this project from just targeting NetCoreAppCurrent to special-casing it for Windows, Unix, and Browser.

I checked with @safern.  The RID graph has all those other platforms as derivatives of Unix, so the Unix target is covering them:
https://github.com/dotnet/runtime/blob/main/src/libraries/OSGroups.json
https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/eng/targetframeworksuffix.props
We probably need to either say that these do work on those platforms (but to my knowledge signal APIs don't... I could be wrong), fix the build to special-case each of those platforms, or do so by moving this into corelib as you suggested.  The already merged PR introduced this, so I won't tackle it here; we can do it subsequently.